math/src/field/traits.rs

```diff
@@ -1,6 +1,42 @@
-use crate::unsigned_integer::traits::IsUnsignedInteger;
+use crate::{fft::errors::FFTError, unsigned_integer::traits::IsUnsignedInteger};
 use std::fmt::Debug;
 
+use super::element::FieldElement;
+
+/// Trait to define necessary parameters for FFT-friendly Fields.
+/// Two-Adic fields are ones whose order is of the form  $2^n k + 1$.
+/// Here $n$ is usually called the *two-adicity* of the field. The
+/// reason we care about it is that in an $n$-adic field there are $2^j$-roots
+/// of unity for every `j` between 1 and n, which is needed to do Fast Fourier.
+/// A two-adic primitive root of unity is a number w that satisfies w^(2^n) = 1
+/// and w^(j) != 1 for every j below 2^n. With this primitive root we can generate
+/// any other root of unity we need to perform FFT.
+pub trait TwoAdicField: IsField {
```

Comment:
Two things I find confusing:
1. While I do prefer the `TwoAdicField` name or those that are named after what they implement rather than a query, they already follow the "question about implementers nature" pattern, so I think we should be consistent and go with `IsTwoAdicField`.
2. Why on callers we redundantly say we want something that is both a `TwoAdicField` and an `IsField`, when the first already implies the latter?

Reply:
If I understood the second question correctly, the reason we require a `TwoAdicField` to implement `IsField` is that users need to specify a root of unity, which is an element of the field itself, so they need to provide its value using the `BaseType`

Reply:
Just for the record (since this is already merged), the second question is about the place where this trait is required, that did `TwoAdicField + IsField` for the trait bounds. Since `TwoAdicField` already has the bound `IsField`, everything after the `+` is redundant.